craterpy/classes.py

```diff
@@ -469,3 +475,186 @@ def plot(self, name="", ax=None, alpha=0.2, **kwargs):
+
+
+    @classmethod
+    def read_shapefile(cls, filename, body="Moon", units="m"):
+        """
+        Read crater data from a shapefile or GeoJSON file.
+        
+        Parameters
+        ----------
+        filename : str
+            Path to the shapefile or GeoJSON file.
+        body : str, optional
+            Planetary body, e.g. Moon, Vesta (default: Moon).
+            If the file contains a 'body' or 'planet' field, that value will be used instead.
+        units : str, optional
+            Length units of radius/diameter, m or km (default: m).
+            If the file contains a 'units' field, that value will be used instead.
+        
+        Returns
+        -------
+        CraterDatabase
+            A new CraterDatabase instance containing the data from the file.
+            
+        Notes
+        -----
+        This method assumes the file was previously created by CraterDatabase.to_geojson()
+        or has a compatible format with lat/lon coordinates and radius or diameter information.
+        
+        If the file contains different coordinate column names than expected, this method
+        will attempt to identify them by common names (e.g., 'lat', 'latitude', 'lon', 'longitude').
+        """
+        gdf = gpd.read_file(filename)
+        
+        # Check if the file has body or units info
+        if 'body' in gdf.columns and gdf['body'].nunique() == 1:
+            body = gdf['body'].iloc[0]
+        elif 'planet' in gdf.columns and gdf['planet'].nunique() == 1:
+            body = gdf['planet'].iloc[0]
+            
+        if 'units' in gdf.columns and gdf['units'].nunique() == 1:
+            units = gdf['units'].iloc[0]
+        
+        # Create a working copy to modify
+        data = gdf.copy()
+        
+        # Try to identify latitude column
+        lat_col = None
+        lat_candidates = ['lat', 'latitude', 'latitude_deg', 'lat_deg', 'y']
+        for col in lat_candidates:
+            if col in data.columns:
+                lat_col = col
+                break
+        
+        # Try to identify longitude column
+        lon_col = None
+        lon_candidates = ['lon', 'longitude', 'longitude_deg', 'lon_deg', 'x']
+        for col in lon_candidates:
+            if col in data.columns:
+                lon_col = col
+                break
+        
+        # If lat/lon columns not found, try to extract from Point geometry
+        if (lat_col is None or lon_col is None) and all(geom.geom_type == 'Point' for geom in data.geometry):
+            data['lon'] = data.geometry.x
+            data['lat'] = data.geometry.y
+            lat_col = 'lat'
+            lon_col = 'lon'
+        
+        # If still no lat/lon, raise error
+        if lat_col is None or lon_col is None:
+            raise ValueError("Could not identify latitude and longitude columns.")
+        
+        # Standardize column names
+        data = data.rename(columns={lat_col: 'lat', lon_col: 'lon'})
```

Comment:
i think we try to preserve the original column names in the CraterDatabase constructor, we should be consistent